sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobUrlParts.java

```diff
@@ -0,0 +1,343 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.blob;
+
+import com.azure.core.implementation.http.UrlBuilder;
+import com.azure.core.implementation.util.ImplUtils;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.storage.blob.specialized.BlobServiceSasQueryParameters;
+import com.azure.storage.common.Constants;
+import com.azure.storage.common.Utility;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.TreeMap;
+
+/**
+ * This class represents the components that make up an Azure Storage Container/Blob URL. You may parse an
+ * existing URL into its parts with the {@link #parse(URL)} class. You may construct a URL from parts by calling {@link
+ * #toUrl()}.
+ */
+public final class BlobUrlParts {
+    private String scheme;
+    private String host;
+    private String containerName;
+    private String blobName;
+    private String snapshot;
+    private BlobServiceSasQueryParameters blobServiceSasQueryParameters;
+    private Map<String, String[]> unparsedParameters;
+
+    /**
+     * Initializes a BlobUrlParts object which helps aid in the construction of a Blob Storage URL.
+     */
+    public BlobUrlParts() {
+        unparsedParameters = new HashMap<>();
+    }
+
+    /**
+     * Gets the URL scheme, ex. "https://".
+     *
+     * @return the URL scheme.
+     */
+    public String getScheme() {
+        return scheme;
+    }
+
+    /**
+     * Sets the URL scheme, ex. "https://".
+     *
+     * @param scheme The URL scheme.
+     * @return the updated BlobUrlParts object.
+     */
+    public BlobUrlParts setScheme(String scheme) {
+        this.scheme = scheme;
+        return this;
+    }
+
+    /**
+     * Gets the URL host, ex. "account.blob.core.windows.net".
```

Comment:
What you're returning is the hostname, not a host. The documentation is misleading.